### PR TITLE
Implemented methods

### PR DESCRIPTION
**Pull request by ValeriyM250**
"put", "get", "size"

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -1,19 +1,50 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+
+    private Object [][] storage = new Object[2][10];
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:8]: https://mate-academy.github.io/jv-program-common-mistakes/java-core/generics/storage.html

**Review by Rommelua — CHANGES_REQUESTED** [review]
Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,62 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private static final int MAX_WIDTH_NUMBER = 2;
+    private Object [][] storage;
```
Comment (andriy-shulepa) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:8]:
```suggestion
    private K[] keys;
    private V[] values;
```

With such approach your solution would be easier to read

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,62 @@
...
+        }
+        if (!haveEquals) {
+            for (int i = 0; i < storage[0].length; i++) {
+                if (storage[0][i] == null && storage[1][i] == null) {
+                    storage[0][i] = key;
+                    storage[1][i] = value;
+                    break;
+                }
+            }
+        }
```
Comment (andriy-shulepa) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:38]: simplify to following approach
1. find index of key
2. set value by index and return
3. set key and value to storage capacity index (you may need to save capacity in class variable)

**Review by andriy-shulepa — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -1,19 +1,55 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.lang.reflect.Array;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ARRAY_LENGTH = 10;
+    private static final int START_INDEX_LENGTH = -1;
+    private static int sizeIndex;
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:9]:
```suggestion
    private int size;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -1,19 +1,55 @@
...
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        keys = (K[]) Array.newInstance(Object.class, MAX_ARRAY_LENGTH);
+        values = (V[]) Array.newInstance(Object.class, MAX_ARRAY_LENGTH);
+        sizeIndex = START_INDEX_LENGTH;
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:16]:
```suggestion
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 6]
```diff
@@ -1,19 +1,55 @@
...
     @Override
     public void put(K key, V value) {
+        if (findKey(key) == -1) {
+            sizeIndex += 1;
+            sizeIndex = (values[sizeIndex] == null) ? sizeIndex : sizeIndex + 1;
+            keys[sizeIndex] = key;
+            values[sizeIndex] = value;
+        } else {
+            values[findKey(key)] = value;
+        }
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:28]: call findKeyIndex only once

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 7]
```diff
@@ -1,19 +1,55 @@
...
     }
 
     @Override
     public V get(K key) {
-        return null;
+        return (findKey(key) == -1) ? null : values[findKey(key)];
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:33]: call findKeyIndex only once

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 8]
```diff
@@ -1,19 +1,55 @@
...
+    }
+
+    private int findKey(K key) {
+        Integer keyIndex = -1;
+        for (int i = 0; i < keys.length; i++) {
+            if (key == keys[i] || keys[i] != null && keys[i].equals(key)) {
+                return keyIndex = i;
+            }
+        }
+        return keyIndex;
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:53]:
```suggestion
    private int findKeyIndez(K key) {
        for (int i = 0; i < keys.length; i++) {
            if (key == keys[i] || keys[i] != null && keys[i].equals(key)) {
                return i;
            }
        }
        return -1;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 9]
```diff
@@ -1,19 +1,55 @@
...
     }
 
     @Override
     public int size() {
-        return -1;
+        for (int i = 0; i < values.length; i++) {
+            if (keys[i] == null && values[i] == null) {
+                return i;
+            }
+        }
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:42]:
```suggestion
        return size;
```

**Review by aaananas — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 10]
```diff
@@ -1,19 +1,57 @@
...
+    private int size;
+
+    private int keyIndex;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        keys = (K[]) Array.newInstance(Object.class, MAX_ARRAY_LENGTH);
+        values = (V[]) Array.newInstance(Object.class, MAX_ARRAY_LENGTH);
+        size = -1;
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:17]:
```suggestion
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 11]
```diff
@@ -1,19 +1,57 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.lang.reflect.Array;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ARRAY_LENGTH = 10;
+    private int size;
+
+    private int keyIndex;
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:10]:
```suggestion
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 12]
```diff
@@ -1,19 +1,57 @@
...
     public void put(K key, V value) {
+        keyIndex = findKey(key);
+        if (keyIndex == -1) {
+            size += 1;
+            size = (values[size] == null) ? size : size + 1;
+            keys[size] = key;
+            values[size] = value;
+        } else {
+            values[keyIndex] = value;
+        }
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:30]:
```suggestion
        int keyIndex = findKey(key);
        if (keyIndex == -1) {
            keys[size] = key;
            values[size] = value;
            size++;
        } else {
            values[keyIndex] = value;
        }
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 13]
```diff
@@ -1,19 +1,57 @@
...
     }
 
     @Override
     public int size() {
+        for (int i = 0; i < values.length; i++) {
+            if (values[i] == null) {
+                return i;
+            }
+        }
+        return size;
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:46]:
```suggestion
        return size;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 14]
```diff
@@ -1,19 +1,57 @@
...
+    }
+
+    private int findKey(K key) {
+        for (int i = 0; i < keys.length; i++) {
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:50]:
```suggestion
        for (int i = 0; i < size; i++) {
```

**Review by aaananas — CHANGES_REQUESTED** [review]

**Review by Rommelua — APPROVED** [review]

**Pull request closed without merging**